### PR TITLE
Romanchuk/Go to 8 .net

### PR DESCRIPTION
**Pull request by alrom**

### CODE REVIEW

File: OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatusServiceTests.cs [hunk 1]
```diff
@@ -30,21 +30,30 @@ public class StatusServiceTests
     private IMapper mapper;
 
     [SetUp]
-    public void Setup()
+    public async Task Setup()
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatusServiceTests.cs]: (I'm curious) This `Setup()` method returns `Task` and is `async`, but I see no `await` inside the method body, and no returns. Could you provide info, why we need such change?

File: OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatusServiceTests.cs [hunk 2]
```diff
@@ -163,15 +175,10 @@ public async Task Delete_WhenIdIsInvalid_ThrowsArgumentOutOfRangeException()
...
-
-            var institutionStatuses = InstitutionStatusGenerator.Generate(5);
-            context.InstitutionStatuses.AddRange(institutionStatuses);
+        context.Database.EnsureDeleted();
+        context.Database.EnsureCreated();
 
-            context.SaveChanges();
-        }
+        // New version of InMemoryTestDB uses ModelBuilder to initialize data from common OutOfSchool contex.
+        // Don't need to use seeding of elements.
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatusServiceTests.cs]: (Commenting standard functionality) Thanks for the sharing this info, at same time could you reconsider the place? 
For me, it looks that this is a standard functionality, not a "hack" or tricky issue, so we don't need to put standard documentation into our code. 
Moreover, there are `SeedDatabase()` in other test classes, and it will be hard (and inefficient) to add the same comment everywhere.

Usually I suggest to put such knowledge sharing notes about important code changes in the PR comment, and also mention it somewhere in team chat/meetings.

File: OutOfSchool/OutOfSchool.WebApi/Models/Workshops/WorkshopBaseDto.cs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿using System.ComponentModel.DataAnnotations;
 using System.ComponentModel.DataAnnotations.Schema;
-using Castle.Core.Internal;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.IdentityModel.Tokens;
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Models/Workshops/WorkshopBaseDto.cs]: Is this using required? Because no changes were introduced in the code, just new using added, and it seems to be irrelevant (as are irrelevant `identity tokens` and `workshop dto`)

File: OutOfSchool/OutOfSchool.WebApi/Services/RatingService.cs [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ﻿using System.Linq.Expressions;
 using AutoMapper;
-using Castle.Core.Internal;
 using Microsoft.Extensions.Localization;
+using Microsoft.IdentityModel.Tokens;
```
Comment (VadymLevkovskyi) [inline OutOfSchool/OutOfSchool.WebApi/Services/RatingService.cs]: Is this using required? Because no changes were introduced in the code, just new using added, and it seems to be irrelevant

**Review by VadymLevkovskyi — COMMENTED** [review]
No logical issues found, have few comments about unclear usings and testing code

File: OutOfSchool/OutOfSchool.WebApi/Models/Workshops/WorkshopBaseDto.cs (hunk 3 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.WebApi/Models/Workshops/WorkshopBaseDto.cs]: In Microsoft.IdentityModel.Tokens we have extension IsNullOrEmpty() for IEnumerable<T>. We can write this method in our extensions.

File: OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatusServiceTests.cs (hunk 2 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatusServiceTests.cs]: Ok. It's the new behavior for InMemoryDatabase. I didn't seed this DB. I tried to describe why.

File: OutOfSchool/OutOfSchool.WebApi/Models/Workshops/WorkshopBaseDto.cs (hunk 3 above)
Comment (VadymLevkovskyi) [reply OutOfSchool/OutOfSchool.WebApi/Models/Workshops/WorkshopBaseDto.cs]: Thank you. So recently it was in `Castle.Core.Internal`, now in `Microsoft.IdentityModel.Tokens`. 
I still suggest either write own implementation or find a related good library with collections extension methods.

File: OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatusServiceTests.cs (hunk 1 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatusServiceTests.cs]: Sorry. I had async method SeedDatabase before. And I forgot to change it.

File: OutOfSchool/OutOfSchool.WebApi/Models/Workshops/WorkshopBaseDto.cs (hunk 3 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.WebApi/Models/Workshops/WorkshopBaseDto.cs]: Our implementation was added.

File: OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatusServiceTests.cs (hunk 2 above)
Comment (alrom) [reply OutOfSchool/OutOfSchool.WebApi.Tests/Services/StatusServiceTests.cs]: Cleared.

**Review by VadymLevkovskyi — APPROVED** [review]
